### PR TITLE
feat: render Scene for gui in sdl

### PR DESCRIPTION
**Pull request by Njord201**
I added a GUI in Raytracer, the result in ppm file is still generated.

image
This is an example.

### CODE REVIEW

File: include/Renderer.hpp [hunk 1]
```diff
@@ -35,22 +35,28 @@ class Raytracer::Renderer {
         ~Renderer() = default;
 
         /**
-         * @brief render the scene
+         * @brief render the scene in a window with SDL2
```
Comment (AxelF44) [inline include/Renderer.hpp:38]:
```suggestion
         * @brief Render the scene in a window with SDL2.
```

File: include/Renderer.hpp [hunk 2]
```diff
@@ -35,22 +35,28 @@ class Raytracer::Renderer {
...
          *
          */
         void renderScene();
 
+        /**
+         * @brief render the final scene in a .ppm file
```
Comment (AxelF44) [inline include/Renderer.hpp:44]:
```suggestion
         * @brief Render the final scene in a .ppm file.
```

File: include/Renderer.hpp [hunk 3]
```diff
@@ -35,22 +35,28 @@ class Raytracer::Renderer {
...
+         *
+         */
+        void renderFinalScene();
+
         /**
          * @brief write a rgb color in a stream
```
Comment (AxelF44) [inline include/Renderer.hpp:50]:
```suggestion
         * @brief Write a rgb color in a stream.
```

File: src/Core/Renderer.cpp [hunk 4]
```diff
@@ -28,8 +29,75 @@ void Raytracer::Renderer::writeColor(std::ostream& o, const Math::Point3D& color
...
+                Raytracer::Ray r(_camera.getOrigin(), rayDirection);
+                Math::Point3D hit = _primitives.hitPoint(r, _lights);
+
+                SDL_SetRenderDrawColor(ren, hit.x(), hit.y(), hit.z(), 255);
+                SDL_RenderDrawPoint(ren, x, y);
+            }
+        }
+
+        SDL_RenderPresent(ren);
+
```
Comment (mathieurobert1) [inline src/Core/Renderer.cpp:90]: useless \n

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by theophile-jr — APPROVED** [review]
Yes Kylian.

**Pull request merged**